File: SI_01a_data_parsing_server_I/src/main/python/DataParser.py
import json
import xml.etree.ElementTree as ET
import yaml
import csv
from src.main.python.Person import Person

class DataParser:
    def parse_xml(self, file_path):
        tree = ET.parse(file_path)
        root = tree.getroot()

        person = Person(name=root.find('Name').text, hobbies=[hobby.text for hobby in root.find('Hobbies')])
        print("Person object from XML:", person)

    def parse_json(self, file_path):
        with open(file_path, 'r') as file:
            person_dict = json.load(file)
            person = Person(**person_dict)
            print("Person object from JSON:", person)

    def parse_yaml(self, file_path):
        with open(file_path, 'r') as file:
            person_dict = yaml.safe_load(file)
            person = Person(**person_dict)
            print("Person object from YAML:", person)

    def parse_csv(self, file_path):
        with open(file_path, 'r') as file:
            reader = csv.DictReader(file)  # Use DictReader for easier mapping

            for row in reader:
                person = Person(name=row['name'], hobbies=row['hobbies'].split(';'))
                print("Person object from CSV:", person)


if __name__ == "__main__":
    data_parser = DataParser()
    try:
        data_parser.parse_xml("../resources/me.xml")
        data_parser.parse_json("../resources/me.json")
        data_parser.parse_yaml("../resources/me.yaml")
        data_parser.parse_csv("../resources/me.csv")
    except Exception as e:
        print(e)

File: SI_01a_data_parsing_server_I/src/main/java/DataParser.java
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.dataformat.csv.CsvMapper;
import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
import com.thoughtworks.xstream.XStream;

import java.io.File;
import java.io.IOException;

public class DataParser {

    public void parseXML(String filePath) throws IOException {
        XStream xstream = new XStream();
        xstream.allowTypesByWildcard(new String[]{"*"});
        xstream.processAnnotations(Person.class);

        var person = xstream.fromXML(new File(filePath));
        print("XML", person);
    }

    public void parseJSON(String filePath) throws IOException {
        var mapper = new ObjectMapper();
        Person person = mapper.readValue(new File(filePath), Person.class);
        print("JSON", person);
    }

    public void parseYAML(String filepath) throws IOException {
        var mapper = new ObjectMapper(new YAMLFactory());
        var person = mapper.readValue(new File(filepath), Person.class);
        print("YAML", person);
    }
    public void parseCSV(String filePath) throws IOException {
        var csvMapper = new CsvMapper();
        var schema = csvMapper.schemaFor(Person.class).withHeader();
        var reader = csvMapper.readerFor(Person.class).with(schema);
        var person =  reader.<Person>readValues(new File(filePath)).next();
        print("CSV", person);
    }

    private void print(String fileType, Object person) {
        System.out.println("Person object from " + fileType + ":"+person);
    }
}

